pkg/utils/configs/config.go

```diff
@@ -43,10 +64,27 @@ func SetQdrouterdDefaults(m *v1alpha1.Qdrouterd) bool {
 			Http: true,
 		})
 	}
-	if len(m.Spec.InterRouterListeners) == 0 {
-		m.Spec.InterRouterListeners = append(m.Spec.InterRouterListeners, v1alpha1.Listener{
-			Port: 55672,
-		})
+	if m.Spec.DeploymentPlan.Role == v1alpha1.RouterRoleInterior {
+		if len(m.Spec.InterRouterListeners) == 0 {
+			m.Spec.InterRouterListeners = append(m.Spec.InterRouterListeners, v1alpha1.Listener{
+				Port: 55672,
+			})
+		}
+
+		if len(m.Spec.InterEdgeListeners) == 0 {
+			m.Spec.InterEdgeListeners = append(m.Spec.InterEdgeListeners, v1alpha1.Listener{
+				Port: 45672,
+				//    Expose: true,
+				//    SslProfile: "default",
+			})
+		}
+	} else if m.Spec.DeploymentPlan.Role == v1alpha1.RouterRoleEdge {
+		if len(m.Spec.InterEdgeConnectors) == 0 {
+			m.Spec.InterEdgeConnectors = append(m.Spec.InterEdgeConnectors, v1alpha1.Connector{
+				Port: 45672,
```

Comment:
Does it make sense to try and create a default edge uplink connector without knowing the hostname to use?

Reply:
totally agree, removed...